### DISCUSSION TITLE
Why use `async void` for event handlers

### DISCUSSION THREAD

**Discussion by xcaptain** [dotnet/maui#23006] [Q&A]
I'm learning maui, I found that for event handlers, the function signature is `async void`, for example:
```C#
    async void GotoCustom(object sender, EventArgs e)
    {
        await Shell.Current.GoToAsync(nameof(CustomTipPage));
    }
```

I think `async void` is harmful, because the caller can't catch exceptions in it, why not use `async Task` ?

**Comment by pictos** [comment]
@xcaptain this is a .NET/C# thing and not .NET MAUI... You can't have async event handlers, so because of that you need to use `async void`. Just surround it in a try/catch block and you will be safe, if you will

**Reply by xcaptain** [reply]
I know `async void` works, but don't you think `async Task` is better?

in Blazor, event handler is `async Task`, see: https://learn.microsoft.com/en-us/aspnet/core/blazor/components/event-handling?view=aspnetcore-8.0

```razor
<p>
    <button @onclick="UpdateHeading">
        Update heading
    </button>
</p>

@code {
    private string headingValue = "Initial heading";

    private async Task UpdateHeading()
    {
        await Task.Delay(2000);

        headingValue = $"New heading ({DateTime.Now})";
    }
}
```

**Reply by pictos** [reply]
It's an event or a `Func<Task>` ? I think it doesn't matter
